File: lib/infrastructure/database/firestore_service.dart
import 'package:cloud_firestore/cloud_firestore.dart';

class FirestoreService {
  static CollectionReference<Map<String, dynamic>> users =
      FirebaseFirestore.instance.collection('users');

  static CollectionReference<Map<String, dynamic>> userMovieList(String userId) =>
      FirestoreService.users.doc(userId).collection('movieList');

  static DateTime dateTimeFromTimestamp(Timestamp val) =>
      DateTime.fromMillisecondsSinceEpoch(val.millisecondsSinceEpoch);

  static Timestamp dateTimeToTimestamp(DateTime time) =>
      Timestamp.fromMillisecondsSinceEpoch(time.millisecondsSinceEpoch);
}

File: _firebase_backend/functions/src/service/firestore_service.ts
import * as admin from "firebase-admin";
import {firestore} from "../admin";


export class FirestoreService {
  /**
 *
 *
 * @return {CollectionReference}
 */
  public static get usersCollection(): admin.firestore.CollectionReference {
    return firestore.collection("users");
  }

  /**
 *
 *
 * @returns {CollectionReference}
 */
  public static userMovieListCollection(
      userId: string
  ): admin.firestore.CollectionReference {
    return FirestoreService.usersCollection
        .doc(userId)
        .collection("movieList");
  }
   /**
 *
 *
 * @returns {CollectionReference}
 */
  public static userMovieRecommendationCollection(
      userId: string
  ): admin.firestore.CollectionReference {
    return FirestoreService.usersCollection
        .doc(userId)
        .collection("contentRecommendations");
  }
}
